bigquery/google/cloud/bigquery/table.py

```diff
@@ -279,6 +279,53 @@ def to_api_repr(self):
             "tableId": self._table_id,
         }
 
+    def to_bqstorage(self, selected_fields=None):
+        """Construct a BigQuery Storage API representation of this table.
+
+        Args:
+            selected_fields (Sequence[ \
+                google.cloud.bigquery.schema.SchemaField, \
+            ]):
+                Optional. A subset of columns to select from this table.
+
+        Returns:
+            Tuple[ \
+                google.cloud.bigquery_storage_v1beta1.types.TableReference, \
+                google.cloud.bigquery_storage_v1beta1.types.TableModifiers, \
+                google.cloud.bigquery_storage_v1beta1.types.TableReadOptions, \
+            ]:
+                A reference to this table in the BigQuery Storage API.
+        """
+        from google.cloud import bigquery_storage_v1beta1
+
+        table_ref = bigquery_storage_v1beta1.types.TableReference()
+        table_ref.project_id = self._project
+        table_ref.dataset_id = self._dataset_id
+
+        modifiers = bigquery_storage_v1beta1.types.TableModifiers()
+        read_options = bigquery_storage_v1beta1.types.TableReadOptions()
+        table_id = self._table_id
+        partition = None
+
+        if "@" in table_id:
```

Comment:
Potential issue here:  table@t1-t2 is the range format, and table@t1 is the snapshot format.  Perhaps a more strict check to ensure you're working with a supported snapshot format?

https://cloud.google.com/bigquery/table-decorators

Reply:
Would `table@t1-t2` work for `tabledata.list`? Would we expect anyone to use the range format for table operations?

Reply:
FYI: I've updated the sample to just strip these out, since they aren't relevant to the actual BigQuery Storage API TableReference. Added a comment in the function docs recommending using the TableModifiers manually.